cpp/oneapi/dal/detail/array_impl.hpp

```diff
@@ -317,29 +327,194 @@ class array_impl : public base {
         });
     }
 
+    const shared* get_if_shared() const noexcept {
+        return std::get_if<shared>(&data_owned_);
+    }
+
+    std::variant<cshared, shared> data_owned_;
+    std::int64_t count_;
+#ifdef ONEDAL_DATA_PARALLEL
+    std::optional<data_parallel_policy> dp_policy_;
+#endif
+};
+
+} // namespace v2
+
+namespace v1 {
+
+template <typename T>
+class array_impl : public base {
```

Comment:
Why you place the compatibility interface of `array` in `array_compat.hpp` but the compatibility interface of `array_impl` you keep here? I think it is better to group them in one file or in one directory

Reply:
It's not better from the user's point of view. The person is going to read code in the headers might be confused by the v1 and v2.